staging/src/k8s.io/apiserver/pkg/server/patch_genericapiserver.go

```diff
@@ -100,3 +90,45 @@ func WithLateConnectionFilter(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+// WithNonReadyRequestLogging rejects the request until the process has been ready once.
+func WithNonReadyRequestLogging(handler http.Handler, hasBeenReadyCh <-chan struct{}) http.Handler {
+	var nonReadyRequestReceived atomic.Bool
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if hasBeenReadyCh != nil {
+			if _, closed := <-hasBeenReadyCh; closed {
```

Comment:
I think it will block when the `hasBeenReadyCh` wasn't closed

Reply:
it does. Fixing.

Reply:
fixed